lib/bloc_provider/lib/src/bloc_provider.dart

```diff
@@ -28,11 +28,13 @@ class BlocProvider<T extends BlocBase> extends StatefulWidget {
     final type = _typeOf<BlocProvider<T>>();
     // ignore: invalid_assignment
     final provider = context.findAncestorWidgetOfExactType<BlocProvider<T>>();
-    assert(provider != null,
-        """A BlocProvider ancestor with Type $type should be given in the widget tree.
+    if (provider == null) {
+      throw Exception(
+          """A BlocProvider ancestor with Type $type should be given in the widget tree.
```

Comment:
Any other ideas or is throwing an exception in this case okay?

Reply:
It's okay